diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Defines the Card Component in Telemetry.
+ * Imported by telemetry.module.ts
+ */
+
+import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { DpslTypes } from '@common/dpsl';
+import { TelemetryService } from 'src/app/core/services/telemetry.service';
+import { TelemetryInfoType, ResponseErrorInfoMessage } from '@common/message';
+import { refreshIntervals } from '../../core/config/data-refresh-intervals';
+
+@Component({
+  selector: 'app-telemetry-card',
+  templateUrl: './telemetry-card.component.html',
+  styleUrls: ['./telemetry-card.component.css'],
+})
+export class TelemetryCardComponent implements OnInit, OnDestroy {
+  @Input({ required: true }) type: TelemetryInfoType | undefined;
+
+  private _intervalId!: number; // the interval id of this card component
+  private _info!: DpslTypes; // the telemetry info received by calling telemetry service
+  private _hasError: boolean = false; // true if it receives an error
+  private _error!: ResponseErrorInfoMessage; // the error message received
```

Comment:
Could we have an optional error type and use it to check if any error is happened?

Reply:
Done!